cljfmt/src/cljfmt/core.cljc

```diff
@@ -386,6 +386,27 @@
 (defn remove-trailing-whitespace [form]
   (transform form edit-all trailing-whitespace? zip/remove))
 
+(defn- replace-with-whitespace [zloc]
+  (zip/replace zloc (whitespace 1)))
+
+(defn- newline? [zloc]
+  (some-> zloc z/tag (= :newline)))
+
+(defn- prev-indenting?
+  "Returns true if all prev elements in line (if any) are whitespaces"
+  [zloc]
+  (or (newline? zloc)
+      (comment? zloc)
+      (and (whitespace? zloc)
+           (recur (zip/prev zloc)))))
+
+(defn- non-indenting-spaces? [zloc]
```

Comment:
This function could be written more succinctly as:

```clojure
(defn- non-indenting-space? [zloc]
  (and (whitespace? zloc) (not (indentation? zloc))))
```

Also we probably want to call it `non-indenting-whitespace?` come to think of it.